mne/io/meas_info.py

```diff
@@ -47,7 +49,76 @@ def _summarize_str(st):
     return st[:56][::-1].split(',', 1)[-1][::-1] + ', ...'
 
 
-class Info(dict):
+class _ChannelNameList(collections.Sequence):
+    """A read-only list, used to provide convenient acces to channel names.
+
+    This list is linked to an Info object. Any changes to the `chs` field of
+    this object are automatically reflected by this list.
+
+    Parameters
+    ----------
+    info : instance of Info
+        The Info structure containing the channel list.
+    """
+    def __init__(self, info):
+        self._channels = info['chs']
+
+    def __getitem__(self, index):
+        """Retrieve the name of the channel with the given index"""
+        if isinstance(index, slice):
+            return [ch['ch_name'] for ch in self._channels[index]]
+        else:
+            return self._channels[index]['ch_name']
+
+    def __len__(self):
+        """Length of the list"""
+        return len(self._channels)
+
+    def __eq__(self, other):
+        """Test for equality"""
+        if isinstance(other, _ChannelNameList):
+            return list(self) == list(other)
+        elif isinstance(other, list):
+            return list(self) == other
+        else:
+            raise ValueError('Cannot compare _ChannelNameList and %s'
+                             % type(other))
+
+    def __ne__(self, other):
+        """Test for non-equality"""
+        return not self.__eq__(other)
+
+    def __repr__(self):
+        """String representation"""
+        if len(self) < 10:
+            channels = ', '.join(self)
+        else:
+            channels = ', '.join(self[:5]) + ' ... ' + ', '.join(self[-5:])
+
+        return '<ChannelNameList | %d channels | %s>' % (len(self), channels)
+
+    def __add__(self, other):
+        """Return a list containing the channels names in both lists"""
+        return list(self) + list(other)
+
+    # Raise descriptive error when the user tries to modify this list.
+    _read_only_error = ("This list of channel names is read-only. It is "
+                        "automatically computed by the info object. "
+                        "Instead of modifying this list, make your "
+                        "modifications to the list of channels in the info "
+                        "object (info['chs']).")
+
+    def __setitem__(self, index, value):
+        raise RuntimeError(_ChannelNameList._read_only_error)
+
+    def __iadd__(self, value):
+        raise RuntimeError(_ChannelNameList._read_only_error)
+
+    def append(self, value):
+        raise RuntimeError(_ChannelNameList._read_only_error)
+
+
+class Info(collections.MutableMapping):
```

Comment:
Now that I think about it, changing the class of `Info` will break writing with h5io, won't it? Because it won't think of it as a `dict` anymore?